### ISSUE TITLE
JSON parse failures because Filebeat sends invalid JSON

### ISSUE THREAD

**Issue by mgm-oss** [elastic/filebeat#297]
We are migrating from Logstash Forwarder to Filebeat and get JSON parse failures when the log messages span multiple lines. We already log in JSON format but Filebeat seams to strip off the first part of the JSON message.

Here's a log message in JSON format as it is logged by the application:

```
{"@timestamp":"2015-11-27T10:08:55.568+0100","@message":"Toolbox - caught following exception AuthenticationException","@source":"foo.ws.error.ApplicationExceptionHandler","@source_host":"myhost","@fields":{"timestamp":1448615335568,"level":"ERROR","line_number":28,"class":"foo.ws.error.ApplicationExceptionHandler","method":"toResponse","exception_class":"foo.ws.authentication.AuthenticationException","exception_message":"Invalid authentication token","stacktrace":"foo.ws.authentication.AuthenticationException: Invalid authentication token\r\n\tat foo.ws.authentication.AuthenticationInterceptor.invoke(AuthenticationInterceptor.java:28)\r\n\tat sun.reflect.GeneratedMethodAccessor1332.invoke(Unknown Source)\r\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\r\n\tat java.lang.reflect.Method.invoke(Method.java:497)\r\n\tat org.jboss.weld.interceptor.reader.SimpleInterceptorInvocation$SimpleMethodInvocation.invoke(SimpleInterceptorInvocation.java:74)\r\n\tat org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNext(AbstractInterceptionChain.java:124)\r\n\tat org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNextInterceptor(AbstractInterceptionChain.java:102)\r\n\tat org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.executeInterception(InterceptorMethodHandler.java:43)\r\n\tat org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.invoke(InterceptorMethodHandler.java:36)\r\n\tat org.jboss.weld.bean.proxy.CombinedInterceptorAndDecoratorStackMethodHandler.invoke(CombinedInterceptorAndDecoratorStackMethodHandler.java:51)\r\n\tat foo.portal.backend.authentication.AuthenticationResource$Proxy$_$$_WeldSubclass.logoff(Unknown Source)\r\n\tat foo.portal.backend.authentication.AuthenticationResource$Proxy$_$$_WeldClientProxy.logoff(Unknown Source)\r\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\r\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\r\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\r\n\tat java.lang.reflect.Method.invoke(Method.java:497)\r\n\tat org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:137)\r\n\tat org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:296)\r\n\tat org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:250)\r\n\tat org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:237)\r\n\tat org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:356)\r\n\tat org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:179)\r\n\tat org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:220)\r\n\tat org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)\r\n\tat org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)\r\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\r\n\tat io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)\r\n\tat io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:61)\r\n\tat io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)\r\n\tat org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)\r\n\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\r\n\tat io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)\r\n\tat io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:56)\r\n\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\r\n\tat io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:45)\r\n\tat io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:63)\r\n\tat io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:58)\r\n\tat io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:70)\r\n\tat io.undertow.security.handlers.SecurityInitialHandler.handleRequest(SecurityInitialHandler.java:76)\r\n\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\r\n\tat org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)\r\n\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\r\n\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\r\n\tat io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:261)\r\n\tat io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:247)\r\n\tat io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:76)\r\n\tat io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:166)\r\n\tat io.undertow.server.Connectors.executeRootHandler(Connectors.java:197)\r\n\tat io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:759)\r\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)\r\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)\r\n\tat java.lang.Thread.run(Thread.java:745)\r\n"},"@tags":["UNKNOWN"]}
```

We activated logging on the Filebeat side. Here's what Filebeat sends:

```
{"@metadata":{"beat":"filebeat","type":"json"},"@timestamp":"2015-11-27T09:09:04.705Z","beat":{"hostname":"myhost","name":"myhost"},"count":1,"fields":null,"input_type":"log","message":"eInterceptorInvocation.java:74)\\r\\n\\tat org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNext(AbstractInterceptionChain.java:124)\\r\\n\\tat org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNextInterceptor(AbstractInterceptionChain.java:102)\\r\\n\\tat org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.executeInterception(InterceptorMethodHandler.java:43)\\r\\n\\tat org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.invoke(InterceptorMethodHandler.java:36)\\r\\n\\tat org.jboss.weld.bean.proxy.CombinedInterceptorAndDecoratorStackMethodHandler.invoke(CombinedInterceptorAndDecoratorStackMethodHandler.java:51)\\r\\n\\tat foo.portal.backend.authentication.AuthenticationResource$Proxy$_$$_WeldSubclass.logoff(Unknown Source)\\r\\n\\tat foo.portal.backend.authentication.AuthenticationResource$Proxy$_$$_WeldClientProxy.logoff(Unknown Source)\\r\\n\\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\\r\\n\\tat sun.reflect.NativeMethodAccessorImpl.invokeeInterceptorInvocation.java:74)\\r\\n\\tat org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNext(AbstractInterceptionChain.java:124)\\r\\n\\tat org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNextInterceptor(AbstractInterceptionChain.java:102)\\r\\n\\tat org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.executeInterception(InterceptorMethodHandler.java:43)\\r\\n\\tat org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.invoke(InterceptorMethodHandler.java:36)\\r\\n\\tat org.jboss.weld.bean.proxy.CombinedInterceptorAndDecoratorStackMethodHandler.invoke(CombinedInterceptorAndDecoratorStackMethodHandler.java:51)\\r\\n\\tat foo.portal.backend.authentication.AuthenticationResource$Proxy$_$$_WeldSubclass.logoff(Unknown Source)\\r\\n\\tat foo.portal.backend.authentication.AuthenticationResource$Proxy$_$$_WeldClientProxy.logoff(Unknown Source)\\r\\n\\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\\r\\n\\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\\r\\n\\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\\r\\n\\tat java.lang.reflect.Method.invoke(Method.java:497)\\r\\n\\tat org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:137)\\r\\n\\tat org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:296)\\r\\n\\tat org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:250)\\r\\n\\tat org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:237)\\r\\n\\tat org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:356)\\r\\n\\tat org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:179)\\r\\n\\tat org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:220)\\r\\n\\tat org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)\\r\\n\\tat org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)\\r\\n\\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\\r\\n\\tat io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)\\r\\n\\tat io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:61)\\r\\n\\tat io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)\\r\\n\\tat org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)\\r\\n\\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\\r\\n\\tat io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)\\r\\n\\tat io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:56)\\r\\n\\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\\r\\n\\tat io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:45)\\r\\n\\tat io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:63)\\r\\n\\tat io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:58)\\r\\n\\tat io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:70)\\r\\n\\tat io.undertow.security.handlers.SecurityInitialHandler.handleRequest(SecurityInitialHandler.java:76)\\r\\n\\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\\r\\n\\tat org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)\\r\\n\\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\\r\\n\\tat io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)\\r\\n\\tat io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:261)\\r\\n\\tat io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:247)\\r\\n\\tat io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:76)\\r\\n\\tat io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:166)\\r\\n\\tat io.undertow.server.Connectors.executeRootHandler(Connectors.java:197)\\r\\n\\tat io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:759)\\r\\n\\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)\\r\\n\\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)\\r\\n\\tat java.lang.Thread.run(Thread.java:745)\\r\\n\"},\"@tags\":[\"UNKNOWN\"]}","offset":53767149,"source":"F:\\wildfly-8.2.0.Final\\WildFly-HOST\\servers\\myhost-node1\\log\\logstash.log","type":"json"}
```

You can see that the `message` field starts somewhere in the middle of the original JSON that was logged and that other fields disappear.

We use Filebeat 1.0.0-rc-2 on Windows Server 2012 R2 Standard.

Here's the Filebeat config:

```yaml
filebeat:
  prospectors:
    -
      paths:
        - F:\wildfly-8.2.0.Final\WildFly-HOST\servers\myhost-node1\log\logstash.log
        - F:\wildfly-8.2.0.Final\WildFly-HOST\servers\myhost-node2\log\logstash.log
      encoding: utf-8
      input_type: log
      ignore_older: 2h
      document_type: json
      scan_frequency: 10s
      force_close_files: true

  registry_file: "C:/ProgramData/filebeat/registry"

output:
  logstash:
    hosts: ["mylogginghost:5000"]
  file:
    path: "C:/ProgramData/filebeat/output"
    filename: logstash-output.log
    rotate_every_kb: 10000
    number_of_files: 7

logging:
  to_files: true
  files:
    name: filebeat.log
    rotateeverybytes: 10485760 # = 10MB
    keepfiles: 7
  level: debug
```

**Comment by unguiculus** [comment]
I created this ticket and only realized afterwards that I was logged in with our company account. I just wanted to show up here in person as well. Thanks.

**Comment by ruflin** [comment]
@unguiculus Can you try if the same issue still exist when you use version 1.0.0? We had someone reporting a similar issue with rc2 and in 1.0.0 it was gone.

**Comment by unguiculus** [comment]
I did not realize 1.0.0 had already been released since neither the release itself nor a release tag show up on Github. Anyways, I just updated and the problem seems to be fixed. Thanks.

**Comment by urso** [comment]
Looks like duplicate of #258. There is a known bug in rc2 failing to correctly send long lines. This should be fixed in recent 1.0.0 release.

**Comment by ruflin** [comment]
@unguiculus Thanks for making us aware of that. Tags were made but not published. Now all the release tags should be available: https://github.com/elastic/filebeat/releases
